components/file_preview_modal/file_preview_modal_image_controls/file_preview_modal_image_controls.tsx

```diff
@@ -0,0 +1,141 @@
+}
+
+const FilePreviewModalImageControls = ({toolbarZoom, setToolbarZoom}: Props) => {
+    // Initial variables and constants
+    // zoom text
+    const [zoomText, setZoomText] = useState<string>();
+    const [selectedZoomValue, setSelectedZoomValue] = useState<ZoomValue | 'customZoom'>();
+
+    // eslint-disable-next-line react/jsx-max-props-per-line
+    const plusSign = <PlusIcon size={16} color='currentColor'/>;
+    // eslint-disable-next-line react/jsx-max-props-per-line
+    const minusSign = <MinusIcon size={16} color='currentColor'/>;
+
+    // Initialize dropdown values
+    const {formatMessage} = useIntl();
+    const autoText = formatMessage({id: 'imageToolbarZoomDropdown.automatic', defaultMessage: 'Automatic'});
+    const fitWidthText = formatMessage({id: 'imageToolbarZoomDropdown.fitWidth', defaultMessage: 'Fit width'});
+    const fitHeightText = formatMessage({id: 'imageToolbarZoomDropdown.fitHeight', defaultMessage: 'Fit height'});
+
+    const zoomLevels = new Map();
+    zoomLevels.set('A', {text: autoText, type: 'auto'});
+    zoomLevels.set('W', {text: fitWidthText, type: 'auto'});
+    zoomLevels.set('H', {text: fitHeightText, type: 'auto'});
+    zoomLevels.set('1', {text: '100%', type: 'scale'});
+    zoomLevels.set('1.25', {text: '125%', type: 'scale'});
+    zoomLevels.set('1.5', {text: '150%', type: 'scale'});
+    zoomLevels.set('2', {text: '200%', type: 'scale'});
+    zoomLevels.set('3', {text: '300%', type: 'scale'});
+    zoomLevels.set('4', {text: '400%', type: 'scale'});
+    zoomLevels.set('5', {text: '500%', type: 'scale'});
+
+    const zoomLevelOptions = [];
+    for (const [zoomLevelKey, zoomLevel] of zoomLevels) {
+        zoomLevelOptions.push(
+            <option
+                key={zoomLevelKey}
+                value={zoomLevelKey}
+            >
+                {zoomLevel.text}
+            </option>,
+        );
+    }
+
+    zoomLevelOptions.push(
+        <option
+            key={'customZoom'}
+            value='customZoom'
+            hidden={true}
+        >
+            {zoomText}
+        </option>,
+    );
+
+    // Handlers
+    // change type to proper type in the future
+    const handleZoomDropdown = (event: ChangeEvent<HTMLSelectElement | HTMLInputElement>) => {
+        const zoomLevel = event.target;
+        if (zoomLevels.get(zoomLevel.value).type === 'scale') {
+            setToolbarZoom(parseFloat(zoomLevel.value));
+        } else {
+            setToolbarZoom(zoomLevel.value as ZoomValue);
+        }
+    };
+
+    const makeZoomHandler = (delta: number) => () => {
+        let newToolbarZoom = typeof toolbarZoom === 'string' ? zoom : toolbarZoom;
+        newToolbarZoom = Math.round(newToolbarZoom * 10) / 10;
+        newToolbarZoom = clamp(newToolbarZoom + delta, minZoom, 5);
+        setToolbarZoom(newToolbarZoom === minZoom ? 'A' : newToolbarZoom);
+    };
+
+    // Callbacks
+    useEffect(() => {
+        if (typeof toolbarZoom === 'number') {
```

Comment:
what about options `'W'` and `'H'`?

Reply:
Oh 🤦 
I had temporarily put this there to avoid lint errors while trying to find a better solution